### ISSUE TITLE
Quarkus does not exposed Jax-rs Ressource which is on an other Maven Module

### ISSUE THREAD

**Issue by BaptisteMeynier** [quarkusio/quarkus#1728]
I'm trying to develop a quarkus application with a Hexagonal architecture.
Application code is available in [github](https://github.com/BaptisteMeynier/Quarkus-Portfolio)
I have 4 modules; Business, Persistence, Webservice and the application is packaged in a global module where I will produce my Quarkus application.

When I launch:
mvn clean package -Pnative
and then my native image
portfolio-app/target/portfolio-app-1.0-SNAPSHOT-runner

Quarkus is not able to expose my PortfolioEndpoint which is not in the main module.
I could put my endpoint in my portfolio-app and it's work, but I don't want to break the hexagonal architecture.
Should have put all my Quarkus features in the same maven module or is it possible to split features in multiple package ?

**Comment by gsmet** [comment]
Could you try to add an empty `META-INF/beans.xml` file in the module containing your endpoints?

**Comment by geoand** [comment]
When I tried this, none of the modules where getting indexed, so not only would JAX-RS endpoints not be picked up, but nor would any of the other annotated classes.

**Comment by gsmet** [comment]
Yes, right now, I think only the main module will be indexed if you don't add an empty `META-INF/beans.xml` or the Jandex Maven plugin.

**Comment by geoand** [comment]
Although I guess we will have a proper solution down the line, should we be documenting those 2 solutions somewhere (I don't really know where that would be...)?

**Comment by gsmet** [comment]
Yeah, so, the issue is that you end up with next issue: the dev mode does not support multi-module either.

That's definitely something we need to fix soon enough as most of the real life projects are multi-modules.

**Comment by BaptisteMeynier** [comment]
Thx the Jandex Maven plugin work well ! My Resource is discovered.

**Comment by BaptisteMeynier** [comment]
I don't know if there is a link with the previously issue. However I had added the jandex Maven plugin.  I created a producer in my main module portfolio-app which has to instantiate beans located in  others modules. It does'nt seems to be called. I added some logs which never appear [Provider class](https://github.com/BaptisteMeynier/Quarkus-Portfolio/blob/spi/portfolio-app/src/main/java/org/acme)
